api/src/main/java/org/pmiops/workbench/cohortbuilder/FieldSetQueryBuilder.java

```diff
@@ -0,0 +1,320 @@
+    }
+    if (!foundIt) {
+      throw new BadRequestException(
+          "One of value, valueNumber, valueDate, or valueNull must be specified for filter on column " +
+              columnFilter.getColumnName());
+    }
+    sqlBuilder.append(columnFilter.getColumnName());
+    sqlBuilder.append(' ');
+    sqlBuilder.append(columnFilter.getOperator());
+    sqlBuilder.append(" ${");
+    sqlBuilder.append(paramName);
+    sqlBuilder.append("}\n");
+  }
+
+  private void handleInClause(ColumnFilter columnFilter, ColumnConfig columnConfig,
+      StringBuilder sqlBuilder, Map<String, QueryParameterValue> paramMap) {
+    String paramName = "p" + paramMap.size();
+    if (columnFilter.getValue() != null || columnFilter.getValueNumber() != null
+        || columnFilter.getValueDate() != null || columnFilter.getValueNull() != null) {
+      throw new BadRequestException("Can't use in operator with single value filter");
+    }
+    List<BigDecimal> valueNumbers = columnFilter.getValueNumbers();
+    List<String> valueStrings = columnFilter.getValues();
+
+    if (valueNumbers == null || valueNumbers.isEmpty()) {
+      if (valueStrings == null || valueStrings.isEmpty()) {
+        throw new BadRequestException("Can't use in operator without values or valueNumbers");
+      } else {
+        if (!columnConfig.type.equals(ColumnType.STRING)) {
+          throw new BadRequestException("Can't use values with column " + columnConfig.name
+              + " of type " + columnConfig.type);
+        }
+        paramMap.put(paramName, QueryParameterValue.array(valueStrings.toArray(new String[0]),
+            String.class));
+      }
+    } else {
+      if (valueStrings == null || valueStrings.isEmpty()) {
+        if (!columnConfig.type.equals(ColumnType.INTEGER)) {
+          throw new BadRequestException("Can't use valueNumbers with column " + columnConfig.name
+              + " of type " + columnConfig.type);
+        }
+        paramMap.put(paramName, QueryParameterValue.array(
+            valueNumbers.stream().map(BigDecimal::longValue).collect(
+                Collectors.toList()).toArray(new Long[0]),
+            Long.class));
+      } else {
+        throw new BadRequestException("Can't use both values and valueNumbers");
+      }
+    }
+    sqlBuilder.append(columnFilter.getColumnName());
+    sqlBuilder.append(" in unnest(${");
+    sqlBuilder.append(paramName);
+    sqlBuilder.append("})\n");
+
+  }
+
+  private void handleColumnFilter(ColumnFilter columnFilter, TableConfig tableConfig,
+      StringBuilder sqlBuilder, Map<String, QueryParameterValue> paramMap) {
+    if (Strings.isNullOrEmpty(columnFilter.getColumnName())) {
+      throw new BadRequestException("Missing column name for column filter");
+    }
+    ColumnConfig columnConfig = findColumn(tableConfig, columnFilter.getColumnName());
+    if (columnConfig == null) {
+      throw new BadRequestException("Column " + columnFilter.getColumnName() + " does not exist");
+    }
```

Comment:
Should findColumn() throw BadRequestException instead of null?

Reply:
This method has been removed.